### PR TITLE
Remove support for emitting and loading Itanium-64 assemblies

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Itanium is long gone.

### CODE REVIEW

**Review by kg — APPROVED** [review]
LGTM

File: src/coreclr/pal/inc/rt/ntimage.h [hunk 1]
```diff
@@ -230,7 +230,6 @@ typedef struct _IMAGE_FILE_HEADER {
 #define IMAGE_FILE_MACHINE_AM33              0x01d3
 #define IMAGE_FILE_MACHINE_POWERPC           0x01F0  // IBM PowerPC Little-Endian
 #define IMAGE_FILE_MACHINE_POWERPCFP         0x01f1
-#define IMAGE_FILE_MACHINE_IA64              0x0200  // Intel 64
```
Comment (jkotas) [inline src/coreclr/pal/inc/rt/ntimage.h:233]: This should be kept

Comment (kg) [reply src/coreclr/pal/inc/rt/ntimage.h:233]: Yeah if the goal is to represent possible values I agree, and then anything that handles its appearance should stay

File: src/coreclr/vm/peassembly.cpp [hunk 2]
```diff
@@ -50,7 +50,7 @@ static void ValidatePEFileMachineType(PEAssembly *pPEAssembly)
         //
         // Though this bypass kicks in for all Load() flavors, the other Load() flavors did detect cpu-matches through various other code paths that still exist.
         // Or to put it another way, this #ifdef makes the (4.5 only) ValidatePEFileMachineType() a NOP for x64, hence preserving 4.0 compatibility.
-        if (actualMachineType == IMAGE_FILE_MACHINE_I386 || actualMachineType == IMAGE_FILE_MACHINE_IA64)
+        if (actualMachineType == IMAGE_FILE_MACHINE_I386)
```
Comment (jkotas) [inline src/coreclr/vm/peassembly.cpp:53]: The whole `#ifdef TARGET_AMD64` block can be deleted. I do not think we need this compat quirk, and the arch mismatches should be detected much earlier.

File: src/coreclr/inc/iceefilegen.h [hunk 3]
```diff
@@ -62,7 +62,6 @@ typedef HRESULT (__stdcall * PFN_DestroyICeeFileGen)(ICeeFileGen ** ceeFileGen);
 #define ICEE_CREATE_MACHINE_MASK       0x0000FF00  // space for up to 256 machine targets (note: most users just do a bit check, not an equality compare after applying the mask)
 #define ICEE_CREATE_MACHINE_ILLEGAL    0x00000000  // An illegal machine name
 #define ICEE_CREATE_MACHINE_I386       0x00000100  // Create a IMAGE_FILE_MACHINE_I386
-#define ICEE_CREATE_MACHINE_IA64       0x00000200  // Create a IMAGE_FILE_MACHINE_IA64
 #define ICEE_CREATE_MACHINE_AMD64      0x00000400  // Create a IMAGE_FILE_MACHINE_AMD64
```
Comment (jkotas) [inline src/coreclr/inc/iceefilegen.h:65]: Nit: Shift the remaining bits to avoid gap?

**Review by jkotas — APPROVED** [review]
LGTM otherwise

File: src/coreclr/pal/inc/rt/ntimage.h (hunk 1 above)
Comment (jkotas) [reply src/coreclr/pal/inc/rt/ntimage.h:233]: There is a lot of unused definitions in this file. It was copied from Windows.

**Pull request merged**